nullaway/src/main/java/com/uber/nullaway/fixserialization/XMLUtil.java

```diff
@@ -0,0 +1,201 @@
+
+      // Root
+      Element rootElement = doc.createElement("serialization");
+      doc.appendChild(rootElement);
+
+      // Suggest
+      Element suggestElement = doc.createElement("suggest");
+      suggestElement.setAttribute("active", String.valueOf(config.suggestEnabled));
+      suggestElement.setAttribute("enclosing", String.valueOf(config.suggestEnclosing));
+      rootElement.appendChild(suggestElement);
+
+      // Annotations
+      Element annots = doc.createElement("annotation");
+      Element nonnull = doc.createElement("nonnull");
+      nonnull.setTextContent(config.annotationConfig.getNonNull().fullName);
+      Element nullable = doc.createElement("nullable");
+      nullable.setTextContent(config.annotationConfig.getNullable().fullName);
+      annots.appendChild(nullable);
+      annots.appendChild(nonnull);
+      rootElement.appendChild(annots);
+
+      // Output dir
+      Element outputDir = doc.createElement("path");
+      outputDir.setTextContent(config.outputDirectory);
+      rootElement.appendChild(outputDir);
+
+      // Writings
+      TransformerFactory transformerFactory = TransformerFactory.newInstance();
+      Transformer transformer = transformerFactory.newTransformer();
+      DOMSource source = new DOMSource(doc);
+      StreamResult result = new StreamResult(new File(path));
+      transformer.transform(source, result);
+    } catch (Exception e) {
+      throw new RuntimeException("Error happened in writing config." + e);
+    }
+  }
+
+  /**
+   * Locates the node at the given key. Key can be nested in format key_1:key_2:...:key_n.
+   *
+   * @param doc Document object of XML.
+   * @param key Nested key.
+   * @return Node if exists with the hierarchy specified by the key, otherwise {@code null}.
+   */
+  private static Node getNodeFromDocument(Document doc, String key) {
+    if (doc == null || key == null || key.equals("")) {
+      return null;
+    }
+    ArrayList<String> keys = new ArrayList<>(Arrays.asList(key.split(":")));
+    Collections.reverse(keys);
+    NodeList candidates = doc.getElementsByTagName(keys.get(0));
+    for (int i = 0; i < candidates.getLength(); i++) {
+      Node candidate = candidates.item(i);
+      if (hasTheEnclosingParents(candidate, keys)) {
+        return candidate;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if the node's hierarchy matches with the keys.
+   *
+   * @param node Node to examine.
+   * @param keys List of parents simple name.
+   * @return {@code true} if node's matches with keys, {@code false} otherwise.
+   */
+  private static boolean hasTheEnclosingParents(Node node, List<String> keys) {
+    for (String key : keys) {
+      if (node == null || !node.getNodeName().equals(key)) {
+        return false;
+      }
+      node = node.getParentNode();
+    }
+    return true;
+  }
+
+  /** Helper class for setting default values when the key is not found. */
+  static class OrElse<T> {
```

Comment:
I think this class could have a better name.  How about `DefaultXMLValueProvider`?

Reply:
f9b2276e14a8271d68644abff821923bc410e480 and minor bug fix.